crates/aptos/src/move_tool/mod.rs

```diff
@@ -369,6 +369,18 @@ pub struct CompilePackage {
     pub(crate) move_options: MovePackageDir,
 }
 
+impl CompilePackage {
+    pub fn new_for_indexer_testing(package_dir: PathBuf) -> Self {
+        Self {
+            save_metadata: true,
+            included_artifacts_args: IncludedArtifactsArgs {
+                included_artifacts: IncludedArtifacts::All,
+            },
+            move_options: MovePackageDir::new(package_dir),
+        }
+    }
+}
+
```

Comment:
tbh I think this piece of code shouldn't need to know anything about how indexer testing works. Maybe instead of adding a new function here, the better approach is to create a indexer testing utils file that will instantiate all these things with the default params for indexer testing. 

Same feedback on the rest of the places that we do this.